psi4/src/psi4/libmints/twobody.cc

```diff
@@ -112,6 +116,61 @@ TwoBodyAOInt::TwoBodyAOInt(const TwoBodyAOInt &rhs) : TwoBodyAOInt(rhs.integral_
 
 TwoBodyAOInt::~TwoBodyAOInt() {}
 
+// Haser 1989, Equation 7 
+void TwoBodyAOInt::update_density(const std::vector<SharedMatrix>& D) {
+    
+    timer_on("Update Density");
+#pragma omp parallel for
+    for (int M = 0; M < nshell_; M++) {
+        for (int N = 0; N < nshell_; N++) {
+            int m_start = bs1_->shell(M).function_index();
+            int num_m = bs1_->shell(M).nfunction();
+
+            int n_start = bs1_->shell(N).function_index();
+            int num_n = bs1_->shell(N).nfunction();
+            
+            double max_dens = 0.0;
+            for (int m = m_start; m < m_start + num_m; m++) {
+                for (int n = n_start; n < n_start + num_n; n++) {
+                    double val = 0.0;
+                    for (int i = 0; i < D.size(); i++) {
+                        val += std::abs(D[i]->get(m, n));
+                    }
+                    if (val > max_dens) max_dens = val;
+                }
+            }
+            
+            max_dens_shell_pair_[M * nshell_ + N] = max_dens;
+        }
+    }
+    timer_off("Update Density");
+
+}
+
+// Haser 1989 Equations 6 to 14
+bool TwoBodyAOInt::shell_significant_density(int M, int N, int R, int S) {
+
+    // THR in Equation 9
+    double density_threshold = screening_threshold_squared_;
+
+    // Equation 13
+    double Q_MN_sq = shell_pair_values_[N * nshell_ + M];
+    double Q_RS_sq = shell_pair_values_[S * nshell_ + R];
+
+    std::initializer_list<double> block_densities {max_dens_shell_pair_[M * nshell_ + N], max_dens_shell_pair_[R * nshell_ + S], 
```

Comment:
This isn't what I meant when I asked for an initializer list. What I recommend was like [example](https://en.cppreference.com/w/cpp/algorithm/max) `std::max({1, 10, 50, 200})` is the example I had in mind. They don't need to use an initializer list class.